Source/HTTP/iOS/http_ios.mm

```diff
@@ -0,0 +1,144 @@
+    if (FAILED(HCHttpCallRequestGetUrl(m_call, &methodCString, &urlCString)))
+    {
+        HCHttpCallResponseSetNetworkErrorCode(m_call, E_FAIL, 0);
+        CompleteAsync(m_asyncBlock, E_FAIL, 0);
+        return;
+    }
+    
+    NSString* urlString = [[NSString alloc] initWithUTF8String:urlCString];
+    NSURL* url = [NSURL URLWithString:urlString];
+    
+    NSString* methodString = [[NSString alloc] initWithUTF8String:methodCString];
+    
+    NSMutableURLRequest* request = [NSMutableURLRequest requestWithURL:url];
+    [request setHTTPMethod:methodString];
+    
+    uint32_t numHeaders = 0;
+    if (FAILED(HCHttpCallRequestGetNumHeaders(m_call, &numHeaders)))
+    {
+        HCHttpCallResponseSetNetworkErrorCode(m_call, E_FAIL, 0);
+        CompleteAsync(m_asyncBlock, E_FAIL, 0);
+        return;
+    }
+    
+    for (uint32_t i = 0; i<numHeaders; ++i)
+    {
+        char const* headerName;
+        char const* headerValue;
+        if (SUCCEEDED(HCHttpCallRequestGetHeaderAtIndex(m_call, i, &headerName, &headerValue)))
+        {
+            NSString* headerNameString = [[NSString alloc] initWithUTF8String:headerName];
+            NSString* headerValueString = [[NSString alloc] initWithUTF8String:headerValue];
+            
+            [request addValue:headerValueString forHTTPHeaderField:headerNameString];
+        }
+    }
+    
+    uint8_t const* requestBody = nullptr;
+    uint32_t requestBodySize = 0;
+    if (FAILED(HCHttpCallRequestGetRequestBodyBytes(m_call, &requestBody, &requestBodySize)))
+    {
+        HCHttpCallResponseSetNetworkErrorCode(m_call, E_FAIL, 0);
+        CompleteAsync(m_asyncBlock, E_FAIL, 0);
+        return;
+    }
+    
+    if (requestBodySize == 0)
+    {
+        m_sessionTask = [m_session dataTaskWithRequest:request completionHandler:
+                         ^(NSData* data, NSURLResponse* response, NSError* error)
+                         {
+                             std::unique_ptr<ios_http> me{this};
+                             me->completion_handler(data, response, error);
+                         }];
+    }
+    else
+    {
+        NSData* data = [NSData dataWithBytes:requestBody length:requestBodySize];
+        
+        m_sessionTask = [m_session uploadTaskWithRequest:request fromData:data completionHandler:
+                         ^(NSData* data, NSURLResponse* response, NSError* error)
+                         {
+                             std::unique_ptr<ios_http> me{this};
+                             me->completion_handler(data, response, error);
+                         }];
+    }
+    
+    [m_sessionTask resume];
+}
+
+NAMESPACE_XBOX_HTTP_CLIENT_END
+
+void Internal_HCHttpCallPerform(
+    _In_ AsyncBlock* asyncBlock,
+    _In_ hc_call_handle_t call
+)
+{
+    std::unique_ptr<xbox::httpclient::ios_http> httpTask(new xbox::httpclient::ios_http(asyncBlock, call));
+    HCHttpCallSetContext(call, &httpTask);
+    httpTask->initiate_request();
```

Comment:
>initiate_request [](start = 14, length = 16)

If this method fails, you must not release your pointer. This method now needs to return a flag that you can inspect